src/nupic/utils/MovingAverage.cpp

```diff
@@ -33,37 +33,33 @@ using namespace nupic::util;
 
 
 MovingAverage::MovingAverage(UInt wSize, const vector<Real32>& historicalValues)
-    : windowSize_(wSize)
+    : slidingWindow_(wSize, historicalValues)
 {
-  if (historicalValues.size() != 0)
-  {
-    copy(
-      historicalValues.begin() + historicalValues.size() - wSize,
-      historicalValues.end(),
-      back_inserter(slidingWindow_));
-  }
-  total_ = Real32(accumulate(slidingWindow_.begin(), slidingWindow_.end(), 0));
+ auto window = slidingWindow_.getData();
+  total_ = Real32(accumulate(begin(window), end(window), 0));
 }
 
 
-MovingAverage::MovingAverage(UInt wSize) : windowSize_(wSize), total_(0) {}
+MovingAverage::MovingAverage(UInt wSize) : 
+  slidingWindow_(wSize), total_(0) {}
 
 
 Real32 MovingAverage::compute(Real32 newVal)
 {
-  if (windowSize_ == slidingWindow_.size())
-  {
-    total_ -= slidingWindow_.front();
-    slidingWindow_.erase(slidingWindow_.begin()); // pop front element
-  }
 
-  slidingWindow_.push_back(newVal);
+  auto popped = slidingWindow_.append(newVal);
```

Comment:
While I agree & I'd like to remove NEUTRAL_VALUE to make the code more generic 

> code should check whether the slidingWindow_ has reached its capacity

I don't like to enforce this approach, I believe this functionality should be handeled by the sliding window. .Can we come up with a solution? 
I have these alternatives: 
- `void append(T)` + `T getLatestDroppedElement() throws` which will throw a defined runtime exception
- `T append(T)` and we'll ask for `T neutralValue` in the constructor
- `T& append(T)` and we'd return `nullptr` if not yet full
What do you think @mrcslws ?